plugin/storage/es/spanstore/writer.go

```diff
@@ -129,10 +129,26 @@ func (s *SpanWriter) WriteSpan(span *model.Span) error {
 	if err := s.createIndex(spanIndexName, spanMapping, jsonSpan); err != nil {
 		return err
 	}
+	jsonSpan.TagsMap = s.tagsToMap(span.Tags)
+	jsonSpan.Tags = nil
+	jsonSpan.Process.TagsMap = s.tagsToMap(span.Process.Tags)
+	jsonSpan.Process.Tags = nil
 	s.writeSpan(spanIndexName, jsonSpan)
 	return nil
 }
 
+func (s *SpanWriter) tagsToMap(kvs []model.KeyValue) map[string]string {
+	tags := map[string]string{}
+	for _, tag := range kvs {
+		if strings.Contains(tag.Key, ":") {
+			s.logger.Warn("Tag key contains \\':\\', at the query time it will be transformed to \\'.\\'")
```

Comment:
Sorry, in that case the implementation makes sense apart from my question below. The only thing I would like is to have a way to configure the whitelisted/blacklisted tags somehow. The scenario I have in mind is you first would whitelist all of them and then if your service grows enough to have more and more flags you can blacklist some of them, and eventually move to a whitelist if there are too many of them. Though I realize that it's not trivial to reconfigure in a way that will not break indexing. Or maybe it is and just needs to be confirmed? I would think that ES was designed with logs that change over time in mind.

Reply:
Once you reach the limit you are not able to store more data in the given index. You could perhaps change the mapping and reindex but probably nobody would want to do that. 

Blacklisting does not make much sense for me. If you are above the limit it might be very inconvenient to list all blacklisted tags. 

I want something that works OOTB for all users especially with high cardinality tags. The other thing is that kibana is not primarily supported system for us.